ecosystem/typescript/sdk_v2/src/types/index.ts

```diff
@@ -817,3 +820,48 @@ export enum AuthenticationKeyScheme {
    */
   DeriveResourceAccountAddress = 255,
 }
+
+/**
+ * TRANSACTION BUILDER
+ */
+
+export type TransactionData = {
+  secondary_signer_addresses?: Array<HexInput>;
+  fee_payer_address?: HexInput;
+} & (EntryFunctionData | ScriptData | EntryFunctionBCSData);
+
+export type TransactionPayload = {
+  type: "entry_function_payload" | "script_payload";
+  payload: TransactionPayloadEntryFunction | TransactionPayloadScript;
+};
+
+export type AptosRawTransaction = {
+  raw_txn: RawTransaction;
+  secondary_signer_addresses?: Array<AccountAddress>;
+  fee_payer_address?: AccountAddress;
+};
+
+export type EntryFunctionData = {
+  function: MoveStructType;
+  type_arguments: Array<MoveType>;
+  arguments: Array<MoveValue>;
+};
+
+export type ScriptData = {
+  code: string;
+  type_arguments: Array<any>; // should be TypeTag
+  arguments: Array<Uint8Array>;
```

Comment:
Side note, only tangentially relevant:

I just realized that `script` arguments are serialized with a type enum before them because for whatever reason (I think it's the bug Jijun was talking about) we don't have a script transaction arg parser, so it doesn't know ahead of time the layout of the transaction arguments.

We can handle it all internally in the typetag stuff, but this `arguments` field will be an `Array<TransactionArgument>` or at least something we serialize differently, whereas in `EntryFunction` the `arguments` fields don't need to have their types prepended in the serialized args, because it's a "fixed" layout of arguments, so to speak.